### ISSUE TITLE
how to track multiple objects during inference

### ISSUE THREAD

**Issue by HW140701** [yangchris11/samurai#3]
How to track multiple objects during inference?

I added multiple lines of bbox data to first_frame_bbox.txt, but only one box is always tracked during inference.

**Comment by yangchris11** [comment]
The current `demo.py` script is for SOT usage. as you can refer to the code that we only take the very first line from the annotations the bbox prompt.

https://github.com/yangchris11/samurai/blob/5b31709e2c2848c8f1e4aa8570cfac04ced85439/scripts/demo.py#L70-L73

**Comment by HW140701** [comment]
ths.
I modify the code of this line
```python
bbox, track_label = prompts[0]
```
to
```python
bbox, track_label = prompts
```
, but i get this error
![image](https://github.com/user-attachments/assets/cab8ba50-9f10-40b8-9e7b-a63016e57b48)

**Comment by johnsmith0031** [comment]
Many code indicating that samurai only support tracking 1 object now. Any plan to add multiple objects support in the future?
```python
...
if ious[0][best_iou_inds] > self.stable_ious_threshold: # handling only the first object
    self.kf_mean, self.kf_covariance = self.kf.update(self.kf_mean, self.kf_covariance, self.kf.xyxy_to_xyah(high_res_bbox))
    self.stable_frames += 1
...
```

And also, _use_mask_as_output returns 7 results instead of 9, causing error when using mask prompt:
```python
def _use_mask_as_output(self, backbone_features, high_res_features, mask_inputs):
    ...
    return (
        low_res_masks,
        high_res_masks,
        ious,
        low_res_masks,
        high_res_masks,
        obj_ptr,
        object_score_logits,
    )
```

**Comment by yangchris11** [comment]
@johnsmith0031 Your understanding of the current released version is correct. Under the current VOT scenario, we only support tracking one object now with box prompt at the first frame.

**Comment by csparker247** [comment]
I hacked this into the demo script today. It required some (probably terrible) changes to the SAM2 code, but you can see the full set of changes here: https://github.com/educelab/samurai/pull/1/files Buyer beware: I just wanted to do a quick evaluation of SAMURAI, and this was sufficient for that evaluation. I will not merge, maintain, or help anyone with these changes.

**Comment by cplop** [comment]
@csparker247 why not use SAM2 as it is as a SOT and instantiate multiple SAM2 per detected object(either from a detector or foundation model)?

**Comment by csparker247** [comment]
@cplop 

> I just wanted to do a quick evaluation of SAMURAI

**Comment by heipinnuo0902** [comment]
It seems not working. I changed the code for sam_base, the demo code and put multiple bboxes in the txt file. After run it again, not working, still single object tracking.

**Comment by claudiogreco** [comment]
Do you plan to support the tracking of multiple objects?
